apex-validation-rules/main/default/classes/RecordValidator.cls

```diff
@@ -591,13 +589,12 @@ public without sharing class RecordValidator {
             SObject parentRecord = record;
 
             for (Schema.SObjectField field : this.fieldChain) {
-                Schema.DescribeFieldResult fieldDescribe = field.getDescribe();
-                String relationshipName = fieldDescribe.getRelationshipName();
+                Schema.DescribeFieldResult localFieldDescribe = field.getDescribe();
 
-                if (fieldDescribe.getSoapType() != Schema.SoapType.Id) {
-                    return parentRecord.get(fieldDescribe.getName());
+                if (localFieldDescribe.getSoapType() != Schema.SoapType.Id) {
```

Comment:
this is always a refactor that `apex-assist` suggests when local variables shadow instance variables